File: benchmarks/tsb/bench_pop_column.ts
/**
 * Benchmark: popColumn on a 100k-row DataFrame
 */
import { DataFrame, popColumn } from "../../src/index.js";

const ROWS = 100_000;
const WARMUP = 3;
const ITERATIONS = 10;
const a = Array.from({ length: ROWS }, (_, i) => i);
const b = Array.from({ length: ROWS }, (_, i) => i * 2);
const c = Array.from({ length: ROWS }, (_, i) => i * 3);
const df = DataFrame.fromColumns({ a, b, c });

for (let i = 0; i < WARMUP; i++) popColumn(df, "b");
const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) popColumn(df, "b");
const total = performance.now() - start;
console.log(
  JSON.stringify({
    function: "pop_column",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_pop_column.py
"""Benchmark: DataFrame.drop column on a 100k-row DataFrame"""
import json, time
import pandas as pd

ROWS = 100_000
WARMUP = 3
ITERATIONS = 10
df_base = pd.DataFrame({"a": range(ROWS), "b": [i*2 for i in range(ROWS)], "c": [i*3 for i in range(ROWS)]})

for _ in range(WARMUP):
    df_base.drop(columns=["b"])

start = time.perf_counter()
for _ in range(ITERATIONS):
    df_base.drop(columns=["b"])
total = (time.perf_counter() - start) * 1000
print(json.dumps({"function": "pop_column", "mean_ms": total / ITERATIONS, "iterations": ITERATIONS, "total_ms": total}))
